python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py

```diff
@@ -337,11 +337,15 @@ def _update_router_bgp_vrf_mlag_neighbor_cfg(
             bgp_vrf["redistribute"]["connected"] = {"enabled": True, "route_map": "RM-CONN-2-BGP-VRFS"}
 
         interface_name = f"Vlan{vlan_id}"
+        peer_group_name = self.inputs.bgp_peer_groups.mlag_ipv4_underlay_peer.name
+        if self.inputs.bgp_peer_groups.use_separate_mlag_peer_group_for_overlay:
+            peer_group_name = self.inputs.bgp_peer_groups.mlag_ipv4_overlay_peer.name
```

Comment:
create a shared_utils to get this name with the logic defined in the schema comment

Reply:
Done.